smartmoneyconcepts/SMC.py

```diff
@@ -442,255 +444,203 @@ def liquidity(cls, ohlc: DataFrame) -> Series:
+                            if obTop == ohlc.high.iloc[last_btm_index + j]
+                            else obBtm
+                        )
+                        obIndex = (
+                            last_btm_index + j
+                            if obTop == ohlc.high.iloc[last_btm_index + j]
+                            else obIndex
+                        )
 
-                        newOrderBlockInfo = {
-                            "top": boxTop,
-                            "bottom": boxBtm,
-                            "volume": ohlc["volume"].iloc[close_index]
-                            + ohlc["volume"].iloc[close_index - 1]
-                            + ohlc["volume"].iloc[close_index - 2],
-                            "type": "Bear",
-                            "loc": boxLoc,
-                            "loc_number": close_index,
-                            "index": len(bearish_order_blocks),
-                            "oblowvolume": (
-                                ohlc["volume"].iloc[close_index]
-                                + ohlc["volume"].iloc[close_index - 1]
-                            ),
-                            "obhighvolume": ohlc["volume"].iloc[close_index - 2],
-                            "breaker": False,
-                        }
-                        bearish_order_blocks.insert(0, newOrderBlockInfo)
-                        if len(bearish_order_blocks) > maxOrderBlocks:
-                            bearish_order_blocks.pop()
-
-        return bullish_order_blocks, bearish_order_blocks
+                    ob[obIndex] = -1
+                    top[obIndex] = obTop
+                    bottom[obIndex] = obBtm
+                    obVolume[obIndex] = (
+                        ohlc["volume"].iloc[close_index]
+                        + ohlc["volume"].iloc[close_index - 1]
+                        + ohlc["volume"].iloc[close_index - 2]
+                    )
+                    lowVolume[obIndex] = (
+                        ohlc["volume"].iloc[close_index]
+                        + ohlc["volume"].iloc[close_index - 1]
+                    )
+                    highVolume[obIndex] = ohlc["volume"].iloc[close_index - 2]
+
+        ob_series = pd.Series(ob, name="OB")
+        top_series = pd.Series(top, name="Top")
+        bottom_series = pd.Series(bottom, name="Bottom")
+        obVolume_series = pd.Series(obVolume, name="OBVolume")
+        lowVolume_series = pd.Series(lowVolume, name="LowVolume")
+        highVolume_series = pd.Series(highVolume, name="HighVolume")
+        mitigated_index_series = pd.Series(mitigated_index, name="MitigatedIndex")
+
+        return pd.concat(
+            [
+                ob_series,
+                top_series,
+                bottom_series,
+                obVolume_series,
+                lowVolume_series,
+                highVolume_series,
+                mitigated_index_series,
+            ],
+            axis=1,
+        )
```

Comment:
I tested this function myself and this is the output I got:
`    
OB      Top   Bottom  OBVolume      LowVolume   HighVolume  MitigatedIndex

8    0      0.0      0.0       0.0       0.000000       0.0000               0
9    0      0.0      0.0       0.0       0.000000       0.0000               0
10   0      0.0      0.0       0.0       0.000000       0.0000               0
11   0  19250.0  18650.0  749596.0  167057.203125  582538.8125              28
12   0      0.0      0.0       0.0       0.000000       0.0000               0
`

As you can see the OB is 0 although there is still data in the other fields could you correct this please

Reply:
it should be 0 
``                        else:
                            if ohlc.low.iloc[close_index] < bottom[currentOB]:
                                ob[j] = 0

Reply:
it can be
```
                        else:
                            if ohlc.low.iloc[close_index] < bottom[currentOB]:
                                ob[j] = 0
                                top[j] = 0.0
                                bottom[j] = 0.0
                                obVolume[j] = 0.0
                                lowVolume[j] = 0.0
                                highVolume[j] = 0.0
```
if you want

Reply:
> you should use -1 or 1 and then 0 for no order block otherwise how can you tell if it is an order block or not?

no, it not working in that way,I add ob first then removing

Reply:
Im still getting some weird results from this function:
`
OB  Top  Bottom  OBVolume  MitigatedIndex  Percentage

0    0  0.0     0.0       0.0               0           0
1    0  0.0     0.0       0.0               0           0
2    0  0.0     0.0       0.0               0           0
3    0  0.0     0.0       0.0               0           0
4    0  0.0     0.0       0.0               0           0
5    0  0.0     0.0       0.0               0           0
6    0  0.0     0.0       0.0               0           0
7    0  0.0     0.0       0.0               0           0
8    0  0.0     0.0       0.0               0           0
9    0  0.0     0.0       0.0               0           0
10   0  0.0     0.0       0.0               0           0
11   0  0.0     0.0       0.0              28          28
12   0  0.0     0.0       0.0               0           0
`

Im getting 28 in the MitigatedIndex and Percentage when the OB is 0

Reply:
fixed